src/main/java/nl/tudelft/planningstool/api/responses/occurrences/OccurrenceResponse.java

```diff
@@ -48,12 +48,12 @@ public long getEndTime() {
         return parseTime(end, "end");
     }
 
-    private static String parseTime(long time) {
+    private static synchronized String parseTime(long time) {
         Date date = new Date(time);
         return DAY_FORMATTER.format(date) + "T" + HOUR_FORMATTER.format(date);
```

Comment:
I would wrap it into a synchronize block around just the formatter, to prevent unnecessary locks for other (possibly synchronized) methods on your `OccurrenceResponse` and `TimeSlot`.

```java
synchronized(DAY_FORMATTER) {
    return DAY_FORMATTER.format(date) + "T" + HOUR_FORMATTER.format(date);
}
```